lib/rest.js

```diff
@@ -1,20 +1,20 @@
-/* global fetch:false */
-
 'use strict';
+
 const h = require('highland'),
   _ = require('lodash'),
   nodeUrl = require('url'),
-  https = require('https'),
+  { Agent } = require('undici'),
```

Comment:
Node 20 already has undici for its native fetch, but I think this is a good idea to maintain compatibility with other Node versions.